### PR TITLE
feat: set LLB platform

### PR DESCRIPTION
**Pull request by andrewrynhard**
This adds the ability to specify a platform to the LLB. This means the
bldr can now build against ARM.

Signed-off-by: Andrew Rynhard <andrew@andrewrynhard.com>

### CODE REVIEW

File: cmd/llb.go [hunk 1]
```diff
@@ -56,9 +77,8 @@ and outputs buildkit LLB to stdout. This can be used as 'bldr pack ... | buildct
 }
 
 func init() {
-	llbCmd.Flags().StringVarP(&options.Target, "target", "t", "", "Target image to build")
+	llbCmd.Flags().StringVarP(&target, "target", "t", "", "Target image to build")
 	llbCmd.MarkFlagRequired("target") //nolint: errcheck
-	llbCmd.Flags().Var(&options.BuildPlatform, "build-platform", "Build platform")
-	llbCmd.Flags().Var(&options.TargetPlatform, "target-platform", "Target platform")
+	llbCmd.Flags().StringVar(&platform, "platform", "", "Target platform")
```
Comment (smira) [inline cmd/llb.go]: should this have sane default, like `environment.LinuxAmd64`?

File: internal/pkg/pkgfile/build.go [hunk 2]
```diff
@@ -44,7 +46,7 @@ func Build(ctx context.Context, c client.Client, options *environment.Options) (
 	}
 
 	loader := solver.BuildkitFrontendLoader{
-		Context: options.GetVariables(),
+		Context: options.ToolchainPlatform.GetVariables(),
```
Comment (smira) [inline internal/pkg/pkgfile/build.go]: OI wonder if we have to pass platform to buildkit in this function? or it should go with `buildkit` arguments? In the latter case we probably need to add that to the README.

Comment (andrewrynhard) [reply internal/pkg/pkgfile/build.go]: Not sure, it seems to be working as is. I can produce the toolchain build for `aarch64`, so maybe neither?

File: cmd/llb.go (hunk 1 above)
Comment (andrewrynhard) [reply cmd/llb.go]: Fixed.

File: internal/pkg/pkgfile/build.go (hunk 2 above)
Comment (smira) [reply internal/pkg/pkgfile/build.go]: yeah, could be, one easy way to tell is to add a new functional test for some simple-as-possible ARM64 build, we just have several ways to call the build process

**Comment by smira** [comment]
I feel we can close this one now

**Pull request closed without merging**